### PR TITLE
clarify pointer add/sub function safety concerns

### PR DESCRIPTION
**Pull request by jannic**
Ralf Jung made the same changes to the offset functions' documentation
in commit fb089156. As add/sub just call offset, the same limitation
applies here, as well.

I did not copy the whole explanation ("In particular, the resulting pointer may *not* be used to access a different allocated object [...]") because I'd consider that as being too repetitive. The documentation of add/sub already refers to the offset function, so people interested in the details can look it up, there.
But changing 'an object' to 'the same object' is a small change which improves clarity a lot.

### CODE REVIEW

**Review by joshtriplett — CHANGES_REQUESTED** [review]

File: src/libcore/ptr.rs [hunk 1]
```diff
@@ -1255,7 +1255,7 @@ impl<T: ?Sized> *const T {
     /// Behavior:
     ///
     /// * Both the starting and resulting pointer must be either in bounds or one
-    ///   byte past the end of an allocated object.
+    ///   byte past the end of *the same* allocated object.
```
[the remark this answers is not in the document]
Comment (jannic) [reply src/libcore/ptr.rs]: The emphasis was copied from identical sentences on the offset functions. Shall I remove it there, as well?

[the remark this answers is not in the document]
Comment (jannic) [reply src/libcore/ptr.rs]: Removed the emphases in doc comments on offset() in a separate commit.

[the remark this answers is not in the document]
Comment (joshtriplett) [reply src/libcore/ptr.rs]: Ah, I missed that!

It's not *that* important, but thanks for updating those for consistency as well,

**Pull request merged**